crates/ruma-state-res/src/room_version.rs

```diff
@@ -157,6 +177,61 @@ impl RoomVersion {
             strict_canonicaljson: true,
             limit_notifications_power_levels: true,
             extra_redaction_checks: true,
+            allow_knocking: false,
+            restricted_join_rules: false,
+        }
+    }
+
+    #[cfg(feature = "unstable-pre-spec")]
+    pub fn version_7() -> Self {
+        Self {
+            version: RoomVersionId::Version7,
+            disposition: RoomDisposition::Stable,
+            event_format: EventFormatVersion::V3,
+            state_res: StateResolutionVersion::V2,
+            enforce_key_validity: true,
+            special_case_aliases_auth: false,
+            strict_canonicaljson: true,
+            limit_notifications_power_levels: true,
+            extra_redaction_checks: true,
+            allow_knocking: true,
+            restricted_join_rules: false,
+        }
+    }
```

Comment:
Can we change all of these to be associated constants instead, and also use the form `Self { version: VersionX, new_setting: true, ..Self::VERSION_BEFORE }`?

Reply:
Such a good idea!! I feel like I tried consts at some point but don't remember why I didn't use them.